ext/semian/semian.c

```diff
@@ -167,13 +134,11 @@ static VALUE
 update_ticket_count(update_ticket_count_t *tc)
 {
   short delta;
-  struct timespec ts = { 0 };
-  ts.tv_sec = kInternalTimeout;
 
   if (get_max_tickets(tc->sem_id) != tc->tickets) {
     delta = tc->tickets - get_max_tickets(tc->sem_id);
 
-    if (perform_semop(tc->sem_id, kIndexTickets, delta, 0, &ts) == -1) {
+    if (perform_semop(tc->sem_id, kIndexTickets, delta, 0) == -1) {
```

Comment:
This is a *little* racier now because there's no longer a timeout when trying to adjust the count. You could conceivably have exhausted all the tickets and then try to lower the overall ticket count, which could cause this to fail now that there's no timeout. But that seems pretty unlikely.